examples/kitty/src/micropython/fs_helpers.c

```diff
@@ -0,0 +1,130 @@
+#include <microkit.h>
+#include <string.h>
+#include <assert.h>
+#include <stdio.h>
+#include <fs/protocol.h>
+#include "micropython.h"
+#include "fs_helpers.h"
+
+extern char *nfs_share;
+struct sddf_fs_queue *nfs_command_queue;
+struct sddf_fs_queue *nfs_completion_queue;
+
+#define REQUEST_ID_MAXIMUM (SDDF_FS_QUEUE_CAPACITY - 1)
+struct request_metadata {
+    struct sddf_fs_command command;
+    struct sddf_fs_completion completion;
+    bool used;
+    bool complete;
+} request_metadata[SDDF_FS_QUEUE_CAPACITY];
+
+#define NUM_BUFFERS SDDF_FS_QUEUE_CAPACITY * 4
+struct buffer_metadata {
+    bool used;
+} buffer_metadata[SDDF_FS_QUEUE_CAPACITY];
+
+int fs_request_allocate(uint64_t *request_id) {
+    for (uint64_t i = 0; i < NUM_BUFFERS; i++) {
```

Comment:
Use the same type for incrementing everywhere (it's different in `fs_buffer_allocate`)